From db829c91eac1f0701efd3924e8dc10092b5d99ba Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 13 Jan 2020 00:45:22 +0100
Subject: [PATCH] Use multithreaded boost

* boost 1.71.0 provides own version of BoostConfig.cmake and CMake will defer to it
  unless set(Boost_NO_BOOST_CMAKE 1) is set. And when BoostConfig.cmake is used
  it fails to find the single thread version even when available because of this bug
  in 1.71:
  https://github.com/boostorg/boost_install/issues/13

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 common/CMakeLists.txt | 1 -
 imap/CMakeLists.txt   | 1 -
 pop/CMakeLists.txt    | 1 -
 smtp/CMakeLists.txt   | 1 -
 4 files changed, 4 deletions(-)

diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 9ebdffa..aefe603 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -31,7 +31,6 @@ webos_machine_dep()
 
 # These variables can change the default behaviour and need to be set before calling find_package
 set(Boost_USE_STATIC_LIBS OFF)
-set(Boost_USE_MULTITHREADED OFF)
 set(Boost_USE_STATIC_RUNTIME OFF)
 find_package(Boost REQUIRED COMPONENTS regex system filesystem) 
 include_directories(Boost_INCLUDE_DIRS)
diff --git a/imap/CMakeLists.txt b/imap/CMakeLists.txt
index 9ed3cc3..e7d543b 100644
--- a/imap/CMakeLists.txt
+++ b/imap/CMakeLists.txt
@@ -35,7 +35,6 @@ include_directories(inc)
 
 # These variables can change the default behaviour and need to be set before calling find_package
 set(Boost_USE_STATIC_LIBS OFF)
-set(Boost_USE_MULTITHREADED OFF)
 set(Boost_USE_STATIC_RUNTIME OFF)
 find_package(Boost REQUIRED COMPONENTS regex)
 include_directories(Boost_INCLUDE_DIRS)
diff --git a/pop/CMakeLists.txt b/pop/CMakeLists.txt
index 7a8dff9..0e1bdf9 100644
--- a/pop/CMakeLists.txt
+++ b/pop/CMakeLists.txt
@@ -35,7 +35,6 @@ include_directories(inc)
 
 # These variables can change the default behaviour and need to be set before calling find_package
 set(Boost_USE_STATIC_LIBS OFF)
-set(Boost_USE_MULTITHREADED OFF)
 set(Boost_USE_STATIC_RUNTIME OFF)
 find_package(Boost REQUIRED COMPONENTS regex system filesystem)
 include_directories(Boost_INCLUDE_DIRS)
diff --git a/smtp/CMakeLists.txt b/smtp/CMakeLists.txt
index eea1645..f84ae09 100644
--- a/smtp/CMakeLists.txt
+++ b/smtp/CMakeLists.txt
@@ -35,7 +35,6 @@ include_directories(inc)
 
 # These variables can change the default behaviour and need to be set before calling find_package
 set(Boost_USE_STATIC_LIBS OFF)
-set(Boost_USE_MULTITHREADED OFF)
 set(Boost_USE_STATIC_RUNTIME OFF)
 find_package(Boost REQUIRED COMPONENTS regex system filesystem)
 include_directories(Boost_INCLUDE_DIRS)
